prover/src/epoch_service.rs

```diff
@@ -449,34 +461,116 @@ mod tests {
 
     #[tokio::test]
     #[traced_test]
-    async fn test_notify() {
+    async fn test_notify_with_one_receiver() {
         // Test epoch_service is really notifying when an epoch or epoch slice has just changed
-
         let epoch_slice_duration = Duration::from_secs(10);
         let epoch_service = EpochService::try_from((epoch_slice_duration, Utc::now())).unwrap();
-        let notifier = epoch_service.epoch_changes.clone();
+        let mut epoch_changes = epoch_service.epoch_changes();
         let counter_0 = Arc::new(AtomicU64::new(0));
         let counter = counter_0.clone();
 
         let res = tokio::try_join!(
             epoch_service
                 .listen_for_new_epoch()
                 .map_err(|e| AppErrorExt::AppError(e)),
-            // Wait for 3 epoch slices + 100 ms (to wait to receive notif + counter incr)
+            // Wait for 3 epoch slices + 500 ms (to wait to receive notification + counter incr)
             tokio::time::timeout(
                 epoch_slice_duration * 3 + Duration::from_millis(500),
                 async move {
                     loop {
-                        notifier.notified().await;
+                        epoch_changes.changed().await.unwrap();
                         debug!("[Notified] Epoch update...");
                         let _v = counter.fetch_add(1, Ordering::SeqCst);
                     }
-                    // Ok::<(), AppErrorExt>(())
                 }
             )
             .map_err(|_e| AppErrorExt::Elapsed)
         );
         assert!(matches!(res, Err(AppErrorExt::Elapsed)));
         assert_eq!(counter_0.load(Ordering::SeqCst), 3);
     }
+
+    #[tokio::test]
+    #[traced_test]
+    async fn test_notify_with_multiple_receivers_and_borrow_check() {
```

Comment:
Not sure if 2 tests for this are needed. Why not keep only the one with multiple receiver?